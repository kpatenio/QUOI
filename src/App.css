.ant-btn.ant-btn-primary {
  background-color: red;
  border-color: green;
}
.ant-btn.ant-btn-primary:hover {
  background-color: green;
}
body {
  color: green;
}
