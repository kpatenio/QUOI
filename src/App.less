// Import LESS colour variables AND theme variables from antd.
// To see the full list of colours: https://ant.design/docs/spec/colors#Base-Color-Palettes
@import '~antd/es/style/themes/default.less';


// Height, margins, etc.
@headerHeight: 64px;

// COLORS
@blackBlue: #00152a; // default colour of antd header. Keep in case we want to reuse this.

.App, #root {
    height: 100%;
}

.header {
    background-color: @blackBlue;
    .languageUIToggle {
        position: absolute;
        // left: 90%;
        right: 5%;
        top: @headerHeight / 4; // FIXME - refactor. To center the button.
    }
}

.main {
    // TODO - do more research on screen size
    .titleHeader {
        margin-top: 1.2em;
        text-align: center;
        #title {
            font-weight: bold;
            letter-spacing: 0.1em;
            // text-transform:  uppercase; // only enable if we want to change the overall app logo
            font-size: 4em;
            margin-bottom: 0;
        }
        #slogan {
            font-style: italic;
            font-size: 1.4em;
            font-weight: normal;
            margin-bottom: 1.2em;
        }
    }
    .searchContainer {
        transform: translateX(25%);
        .dictionaryToggle {
            display: block;
            background-color: #2f54eb;
        }
        .searchbar {
            width: 50%;
            display: block;
        }
    }
}