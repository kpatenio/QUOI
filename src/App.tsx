import React, {useState, useEffect} from 'react';
import {Layout, Button, Icon, Tooltip, Input} from "antd";
// import logo from './logo.svg';
import FooterComponent from './components/FooterComponent';

import {I18nextProvider, useTranslation, Trans} from 'react-i18next';
import i18next from 'i18next';

// import './App.css';
import './App.less';
import axios from 'axios';
import DOMPurify from 'dompurify';
// import parse from 'html-react-parser'; // if using this, reinstall package
import entry from './mockedAssets/en-fr/entry';

// TODO - use TypeScript interfaces!
// TODO - use SCSS and given classnames to change styling
// TODO - styling remove .cit with #{word_id} and children "&nbsp"?

const App: React.FC = () => {
  const {t, i18n} = useTranslation();
  const {Header, Content} = Layout;
  const {Search} = Input;
  // const content: string = entry.entryContent // temporary content!
  const sadFaceEmoji: string = '&#128542'
  const sanitizer = DOMPurify.sanitize;
  
  const [testState, setTestState] = useState<string>('');

  const [dictCode, setDictCode] = useState<string>('english-french'); // To be sent to and used by API.
  const [toggleText, setToggleText] = useState<string>(t('toggleLabel_en-fr')); 

  const title = '« quoi »';

  const getContent = (word: string | null) => {
    axios.get(`http://127.0.0.1:5000/${dictCode}/${word}`)
    .then((response) => {
      console.log(response);
        if (!response.data.hasOwnProperty('suggestions')) {
          setTestState(response.data.entryContent);
        } else {
          // TODO - use a new component or tag to display list of suggestions
          setTestState(`No results available for '${word}' ${sadFaceEmoji} \n Did you mean ${response.data.suggestions} ?`);
        }
    })
    .catch(() => {
      // TODO - make error call for actual non-mocked calls
      // TODO - this catch is called if 1. server is offline OR 2.
      // TODO - say "unable to search for <word> at this time" instead of offline server
      setTestState(`The server is currently offline. ${sadFaceEmoji} Please try again later.`);
    })
  }

  const handleChangeLanguage = () => {
    if (i18n.language === 'en') {
      i18n.changeLanguage('fr');
    } else {
      i18n.changeLanguage('en');
    }

    // TODO - refactor!!
    if (dictCode === 'english-french') {
      setToggleText(t('toggleLabel_en-fr'))
    } else {
      setToggleText(t('toggleLabel_fr-en'))      
    }
  }

  const handleClickSearch = (word: string | null) => {
    if (word === null || word.trim() === "") {
      setTestState(t('enterWord'))
    } else {
      getContent(word);
    }
  }

  const handleClickToggle = () => {
    // update dictCode
    if (dictCode === 'english-french') {
      setDictCode('french-english')
    } else {
      setDictCode('english-french');
    }

    // update button language toggle label
    // TODO - refactor!
    if (toggleText === t('toggleLabel_en-fr')) {
      setToggleText(t('toggleLabel_fr-en'));
    } else {
      setToggleText(t('toggleLabel_en-fr'))
    }
}


  /* TODO
    Note that antd's ConfigProvider has locale support with it's own translated placeholders. Note that these can usually still be replaced via placeholder prop.
    This depends on the component. For now, let's use i18next! After implementing all translations, we can use ConfigProvider 
  */
  return (
    <div className="App">
      <Header className="header">
        {/*TODO - determine button*/}
        <Button
          // ghost
          // type="link"
          data-testid="languageUIToggle"
          className="languageUIToggle"
          onClick={handleChangeLanguage}
        >
          {t("languageToggle")}
        </Button>
      </Header>
      <Content className="main">
        <hgroup className="titleHeader">
          <h1 id="title"> {title} </h1>
          <h2 id="slogan"> {t("slogan")} </h2>
        </hgroup>
        <div className="searchContainer">
        <Tooltip placement="left" title={t("tooltipTitle")}>
          <Button
            className="dictionaryToggle"
            data-testid="dictionaryToggle"
            onClick={handleClickToggle}
            type="primary"
          >
            {toggleText}
            <Icon type="swap" />
          </Button>
        </Tooltip>
        <Search
          size="large"
          className="searchbar"
          placeholder={t("searchbarPlaceholder")}
          onSearch={handleClickSearch}
        />
        </div>
        <div dangerouslySetInnerHTML={{ __html: sanitizer(testState) }} />
      </Content>
      <FooterComponent/>
    </div>
  );
}

export default App;
