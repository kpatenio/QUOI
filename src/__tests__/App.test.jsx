import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import App from '../App';

// Use MemoryRouter to maintain react router's context.
// See here: https://reacttraining.com/react-router/web/guides/testing

describe("just a test", () => {
    it('should pass', () => {
        expect(true).toStrictEqual(true);
    });

    it('should render App', () => {
        const {container} = render(
        <MemoryRouter>
            <App/>
        </MemoryRouter>
        );
        expect(container.firstChild).toMatchSnapshot();
    })

    it('should render button', () => {
        const {getByTestId} = render(
            <MemoryRouter>
                <App/>
            </MemoryRouter>
            );
        getByTestId('dictionaryToggle');
    })
})