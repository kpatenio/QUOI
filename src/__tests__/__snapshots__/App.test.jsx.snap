// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`just a test should render App 1`] = `
<div
  class="App"
>
  <header
    class="header ant-layout-header"
  >
    <button
      class="ant-btn languageUIToggle"
      data-testid="languageUIToggle"
      type="button"
    >
      <span>
        languageToggle
      </span>
    </button>
  </header>
  <main
    class="main ant-layout-content"
  >
    <hgroup
      class="titleHeader"
    >
      <h1
        id="title"
      >
         
        « quoi »
         
      </h1>
      <h2
        id="slogan"
      >
         
        slogan
         
      </h2>
    </hgroup>
    <div
      class="searchContainer"
    >
      <button
        class="ant-btn dictionaryToggle ant-btn-primary"
        data-testid="dictionaryToggle"
        type="button"
      >
        <span>
          toggleLabel_en-fr
        </span>
        <i
          aria-label="icon: swap"
          class="anticon anticon-swap"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="swap"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M847.9 592H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h605.2L612.9 851c-4.1 5.2-.4 13 6.3 13h72.5c4.9 0 9.5-2.2 12.6-6.1l168.8-214.1c16.5-21 1.6-51.8-25.2-51.8zM872 356H266.8l144.3-183c4.1-5.2.4-13-6.3-13h-72.5c-4.9 0-9.5 2.2-12.6 6.1L150.9 380.2c-16.5 21-1.6 51.8 25.1 51.8h696c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </i>
      </button>
      <span
        class="ant-input-search searchbar ant-input-affix-wrapper ant-input-affix-wrapper-lg"
      >
        <input
          class="ant-input ant-input-lg"
          placeholder="searchbarPlaceholder"
          type="text"
          value=""
        />
        <span
          class="ant-input-suffix"
        >
          <i
            aria-label="icon: search"
            class="anticon anticon-search ant-input-search-icon"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="search"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
              />
            </svg>
          </i>
        </span>
      </span>
    </div>
    <div />
  </main>
  <footer
    class="footer ant-layout-footer"
  >
    <a
      class="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </footer>
</div>
`;
